### ISSUE TITLE
Current Directory Changed Incorrectly Under OpenCover in .Net Core

### ISSUE THREAD

**Issue by adamrodger** [xunit/xunit#1579]
Repo: https://github.com/adamrodger/xunit-directory-bug

When running coverage reports with xUnit and OpenCover under .Net Core, `Directory.GetCurrentDirectory()` returns the path to the xunit executable instead of the bin folder of your project. When running under .Net Framework, this isn't the case.

I notice the code has a number of places where it does `#if NET452` and then  resets the current directory, so I assume this is the problem, like this:

```csharp
#if NET452
    Directory.SetCurrentDirectory(Path.GetDirectoryName(TestAssembly.Assembly.AssemblyPath));
#endif
```

e.g. https://github.com/xunit/xunit/blob/c7140cbc14a0d190931add88c2c0be9f3f2de440/src/xunit.execution/Sdk/Frameworks/Runners/TestAssemblyRunner.cs#L185-187

See #1509

**Comment by bradwilson** [comment]
You cannot rely upon the current directory being correct _unless you set it yourself_.

**Comment by bradwilson** [comment]
The reason this code is gated by `#if NET452` is because there is no way to set the current directory in .NET Standard 1.1.

**Comment by adamrodger** [comment]
They sound like contradictory statements though. I need to set the current
directory myself because xunit has changed it, but sometimes I won't be
able to do that...?

Also, it behaves differently depending on the target framework which can't
be considered correct.

**Comment by bradwilson** [comment]
It makes a best effort to set the current directory, but it will not always be successful (running on non-desktop CLR, or running assemblies in parallel, which mean the current directory cannot be relied upon).

**Comment by adamrodger** [comment]
If the restriction is only on .Net Standard 1.1, couldn't we change the code to `#if !NETSTANDARD1_1`..? That would make the behaviour consistent on every platform where that's possible. I'll give it a try today.

**Comment by bradwilson** [comment]
That library is only built in two configurations: `net452` and `netstandard1.1`.

https://github.com/xunit/xunit/blob/cbed08a46dc1d598dfae9bc80296990169c71eda/src/xunit.execution/xunit.execution.csproj#L9

**Comment by bradwilson** [comment]
Closing as Won't Fix.
